### ISSUE TITLE
Unable to sign in via SAML in Cloudstack Simulator

### ISSUE THREAD

**Issue by elShiaLabeouf** [apache/cloudstack#6365]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
UI: SAML login
~~~

##### CLOUDSTACK VERSION
~~~
main branch
commit: f572c7ab74508366b3b2ccbb2c0e6eeaa872fd36
~~~

##### CONFIGURATION
SAML authentication activated

##### OS / ENVIRONMENT
Environment: development,
Docker + Docker-compose, 
"cloudstack-simulator" container running on localhost:5050/, 
Ruby on Rails server running on localhost:3000/ as custom IdP implemented via the saml_idp gem.

##### SUMMARY
The user is being redirected to login form after a successful SAML SSO login.

##### STEPS TO REPRODUCE
After the IdP sends POST http://localhost:5050/client/api?command=samlSso , it receives a response header `set-cookie: userid=063f7aef-f355-4e4e-85f3-dcbaef02bb84` from the Cloudstack:
![Screenshot from 2022-05-07 11-57-05](https://user-images.githubusercontent.com/20026712/167247777-176b2eb7-0744-44cc-a38e-2dae7ae29bbf.png)

The path in the header is not specified, so the cookie is being set to "/client" Path:
![Screenshot from 2022-05-07 11-57-31](https://user-images.githubusercontent.com/20026712/167247915-07c3a082-c842-4fd6-b585-f9c81d5038c0.png)

Meanwhile, the previously successful POST got a response with the 302 status and the response header `location: http://localhost:5050/`, so the browser is redirected to that location, but instead of a dashboard we see the login form again:
![Screenshot from 2022-05-07 11-57-05](https://user-images.githubusercontent.com/20026712/167248145-5cab61c0-348b-4ec9-93f9-2728212ab666.png)

OK, now we open `localhost:5050/client/` in the browser and fix the "userid" cookie Path to '/' manually: 
![Screenshot from 2022-05-07 11-57-40](https://user-images.githubusercontent.com/20026712/167248201-e908452a-8a4a-44ce-a33f-aa516b6a8911.png)

Then refresh the `localhost:5050/` tab and voila:
![Screenshot from 2022-05-07 11-57-48](https://user-images.githubusercontent.com/20026712/167248229-cb6b7a04-9cb2-4988-bfbd-8731e05e3957.png)

My thoughts:
I'm aware that Cloudstack UI on production is run on `/client` path, so the bug wouldn't reproduce there, but I'm convinced that we shouldn't implicitly hardcode the cookie to the `/client` Path. 

Instead I propose to explicitly set it to the `/` Path.

I'm creating a PR to fix this and get back to you shortly.

P.S. Thank you for the SAML authentication option - it's a very convenient way to authorize users in Cloudstack! 

##### EXPECTED RESULTS

~~~
The user is redirected to his Cloudstack dashboard
~~~

##### ACTUAL RESULTS

~~~
The user is shown the login form
~~~

**Comment by elShiaLabeouf** [comment]
The [PR](https://github.com/apache/cloudstack/issues/6365) is ready, please share your thoughts on that!

**Comment by rohityadavcloud** [comment]
I think it's happening solely due to two different webservers used in development mode. The mgmt server is on port 8080, UI/npm server on 5050. You can try to find a workaround using saml related global settings (for ex. how it is redirecting traffic). I believe, this is not a production issue you're facing (where the static UI and API is served by Jetty).

**Comment by elShiaLabeouf** [comment]
Hi @rohityadavcloud ! Thanks for reaching out on Saturday :)

You're correct, this is not a production issue, I'm using the cloudstack-simulator docker image strictly in development. The image helped me a great deal though - its out-of-box/all-in-one strategy allowed me to quickly set it up in docker-compose and move on to the development itself. Its easiness is really mind-blowing, taking into account the complexity of the project. But unfortunately, SAML SSO refused to work in the "out-of-box mode".

When I found the issue, I realized that I had two options to solve it:
- either the Cloudstack UI URI must be changed to "/client";
- or the cookie should be assigned to "/" Path instead of "/client"

If the first option were easy, I'd've gone with that variant. But neither CS repository surfing nor examining docker image helped me with this. Plus personally, I'm not such a fan of the idea of the hardcoded cookie which forces to use "/client" path for UI if a dev wants to use SAML SSO.
For the second option it was quite easy to find the code section where cookie headers are assigned and then to build a new simulator docker image to test it.

For me the option of using the custom cloudstack-simulator docker image is appropriate - SAML logins in the new containers which use the new docker image work well; plain logins also work well. But I put myself in the place of a next person who is to implement SAML SSO in one's project, and I'd love to save the one from the debugging process I went through.
